src/jobs.py

```diff
@@ -335,6 +337,7 @@ def __init__(self, *args, **kwargs):
         self.memory = None                      # input number
```

Comment:
Can these comments be updated with a some more detail about what the field is meant to be used for! 

As an aside, we are using python 3.5+ so we have the luxury of type setting these variables like so:
```python
self.memory: int = None
self.source_file: str = None
...
```

While this still doesn't give us any runtime guarantees it's certainly an elegant (and now pythonic : )) way of documenting variable types!